lib/samson/hooks.rb

```diff
@@ -63,6 +110,32 @@ def render_views(name, view, *args)
         nil
       end
 
+      def plugin_setup
+        Samson::Hooks.plugins.
+          each(&:require).
+          each(&:precompile_assets).
+          each(&:add_lib_path).
+          each(&:add_decorators)
+      end
+
+      def render_javascripts(view)
+        Samson::Hooks.plugins.map(&:javascripts).flatten.each do |script|
+          view.instance_exec do
+            concat view.javascript_include_tag(view.asset_path(script))
```

Comment:
I'm still a little confused here. This will pull out all the files under `PLUGIN/assets/javascripts/*.js` and create URLs like `samson.com/flowdock_plugin.js` but are they namespaced under the plugin name in case of a filename conflict?
Also, I don't think the filenames in `app/assets` necessarily map to URLs - many assets can be compiled into one via application.js / css and also Rails fingerprinting.

That said, I am probably just being too pedantic and if you are confident that this is the way to go then let's do it :+1: